### PR TITLE
Fix minor issues

### PR DESCRIPTION
**Pull request by atgeirr**
This silences a few warnings, and makes the tests run properly in debug mode.

### CODE REVIEW

**Comment by atgeirr** [comment]
jenkins build this please

File: opm/grid/polyhedralgrid/grid.hh [hunk 1]
```diff
@@ -983,9 +983,13 @@ namespace Dune
...
           }
-        case dim:
+        case 2:
           {
-            return 1;
+            return dim == 2 ? 1 : 0;
+          }
+        case 3:
+          {
+            return dim == 3 ? 1 : 0;
```
Comment (blattms) [inline opm/grid/polyhedralgrid/grid.hh:992]: Why increase the complexity here?

The code does the same as before, seems less explicit, and contains two more switch statements. Please leave as is.

File: opm/grid/polyhedralgrid/grid.hh [hunk 2]
```diff
@@ -1013,10 +1017,19 @@ namespace Dune
...
+              const int coordIndex = GlobalCoordinate :: dimension * seed.index();
+              return copyToGlobalCoordinate( grid_.node_coordinates + coordIndex );
+            }
+          }
+        case 3:
+          {
+            if (dim == 3) {
+              const int coordIndex = GlobalCoordinate :: dimension * seed.index();
+              return copyToGlobalCoordinate( grid_.node_coordinates + coordIndex );
+            }
```
Comment (blattms) [inline opm/grid/polyhedralgrid/grid.hh:1032]: Same thing here. Code is less readable and has more code duplication. What is the reason?

I suggest to leave this as is.

**Review by blattms — CHANGES_REQUESTED** [review]
Unfortunately, you did not tell what the warning is or what the issues where that this fixed.

The const reference change is clear. The rest is not, hence me complaints.

File: opm/grid/polyhedralgrid/grid.hh (hunk 1 above)
Comment (atgeirr) [reply opm/grid/polyhedralgrid/grid.hh:992]: This is the warning I get:
```
In file included from /Users/atgeirr/opm/src/opm-grid/tests/test_polyhedralgrid.cpp:7:
In file included from /Users/atgeirr/opm/src/opm-grid/opm/grid/polyhedralgrid.hh:1:
/Users/atgeirr/opm/src/opm-grid/opm/grid/polyhedralgrid/grid.hh:975:15: warning: no case matching constant switch condition '2'
      switch (codim)
              ^~~~~
/Users/atgeirr/opm/src/opm-grid/opm/grid/polyhedralgrid/grid.hh:1182:27: note: in instantiation of function template specialization 'Dune::PolyhedralGrid<3, 3,
      double>::corners<Dune::PolyhedralGridEntitySeed<2, const Dune::PolyhedralGrid<3, 3, double> > >' requested here
          const int nVx = corners( seed );
                          ^
/Users/atgeirr/opm/src/opm-grid/opm/grid/polyhedralgrid/geometry.hh:131:49: note: in instantiation of function template specialization 'Dune::PolyhedralGrid<3, 3,
      double>::geometryType<Dune::PolyhedralGridEntitySeed<2, const Dune::PolyhedralGrid<3, 3, double> > >' requested here
    GeometryType type () const { return data()->geometryType( storage_.seed() ); }
                                                ^
/Users/atgeirr/opm/src/opm-grid/opm/grid/polyhedralgrid/geometry.hh:123:29: note: in instantiation of member function 'Dune::PolyhedralGridBasicGeometry<1, 3, const
      Dune::PolyhedralGrid<3, 3, double> >::type' requested here
      GeometryType myType = type();
                            ^
/Users/atgeirr/opm/src/opm-grid/opm/grid/polyhedralgrid/geometry.hh:242:7: note: in instantiation of member function 'Dune::PolyhedralGridBasicGeometry<1, 3, const
      Dune::PolyhedralGrid<3, 3, double> >::PolyhedralGridBasicGeometry' requested here
    : Base( data, seed )
      ^
/Users/atgeirr/opm/src/opm-grid/opm/grid/polyhedralgrid/entity.hh:118:24: note: in instantiation of member function 'Dune::PolyhedralGridGeometry<1, 3, const
      Dune::PolyhedralGrid<3, 3, double> >::PolyhedralGridGeometry' requested here
      return Geometry( GeometryImpl( data(), seed_ ) );
                       ^
/Users/atgeirr/duneinstallhack/include/dune/grid/common/entity.hh:143:52: note: in instantiation of member function 'Dune::PolyhedralGridEntityBasic<2, 3, const
      Dune::PolyhedralGrid<3, 3, double> >::geometry' requested here
    Geometry geometry () const { return realEntity.geometry(); }
                                                   ^
/Users/atgeirr/duneinstallhack/include/dune/grid/test/checkidset.hh:103:41: note: in instantiation of member function 'Dune::Entity<2, 3, const Dune::PolyhedralGrid<3, 3,
      double>, PolyhedralGridEntity>::geometry' requested here
              if (! FloatCmp::eq(entity.geometry().center(), idContainer[id], 1e-12 ))
                                        ^
/Users/atgeirr/duneinstallhack/include/dune/grid/test/checkidset.hh:122:5: note: in instantiation of function template specialization
      'Dune::checkInjectivity<Dune::PolyhedralGrid<3, 3, double>, Dune::PolyhedralGridIdSet<3, 3, double> >' requested here
    checkInjectivity(grid, idSet);
    ^
/Users/atgeirr/duneinstallhack/include/dune/grid/test/gridcheck.hh:1022:3: note: in instantiation of function template specialization
      'Dune::checkIdSet<Dune::PolyhedralGrid<3, 3, double>, Dune::PolyhedralGridIdSet<3, 3, double> >' requested here
  checkIdSet(g, g.localIdSet());
  ^
/Users/atgeirr/opm/src/opm-grid/tests/test_polyhedralgrid.cpp:290:9: note: in instantiation of function template specialization
      'gridcheck<Dune::PolyhedralGrid<3, 3, double> >' requested here
        gridcheck( grid );
        ^
```

Comment (atgeirr) [reply opm/grid/polyhedralgrid/grid.hh:992]: Open to better ways to kill it.

Comment (blattms) [reply opm/grid/polyhedralgrid/grid.hh:992]: Shouldn't a default clause like: `default: return 0;` do the trick, too? That should tell the overly smart compiler that we have a sensible default.

Comment (atgeirr) [reply opm/grid/polyhedralgrid/grid.hh:992]: That works, thanks. Not sure why I did not think of it in the first place!

**Comment by atgeirr** [comment]
Update force-pushed.

The test failure is strange: a timeout is reported, but it looks to me like the test case ran, so I think there is a fair chance it was just a fluke.

**Comment by atgeirr** [comment]
jenkins build this please

File: opm/grid/polyhedralgrid/grid.hh (hunk 1 above)
Comment (blattms) [reply opm/grid/polyhedralgrid/grid.hh]: Our brain works in mysterious ways (at least mine often does), once you are on a track...

**Review by blattms — APPROVED** [review]

**Comment by atgeirr** [comment]
Self-merging as it is green and approved.

**Pull request merged**